components/cqrs/name.go

```diff
@@ -0,0 +1,42 @@
+package cqrs
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ObjectName name returns object name in format [package].[type name].
+// It ignores if the value is a pointer or not.
+func ObjectName(v interface{}) string {
+	s := fmt.Sprintf("%T", v)
+	s = strings.TrimLeft(s, "*")
+
+	return s
+}
+
+// StructName name returns struct name in format [type name].
+// It ignores if the value is a pointer or not.
+func StructName(v interface{}) string {
+	segments := strings.Split(fmt.Sprintf("%T", v), ".")
+
+	return segments[len(segments)-1]
+}
+
+type namedMessage interface {
```

Comment:
maybe just Named and public?

Reply:
It does not have to be public as there is no reference to it at all outside of the function.

Reply:
:+1: